xla/service/gpu/fusions/transpose_mlir.cc

```diff
@@ -405,5 +410,376 @@ IndexingMap MlirTransposeFusion::GetIndexing(bool input,
   return result;
 }
 
+namespace {
+
+struct HloBytesUsageDesc {
+  int tile_size;
+  int elem_bytes;
+  int ref_count;
+  int ref_id;
+  int unused_bytes;
+};
+
+std::optional<HloBytesUsageDesc> GetHloBytesUsageDesc(
+    HloInstructionAdaptor& instr_adaptor,
+    absl::InlinedVector<int, 2>& live_range,
```

Comment:
What does this mean? It seems to be a map from ID to ID, but I don't really understand how that is a live range.